plugins/kotlin/gradle/gradle/tests/test/org/jetbrains/kotlin/idea/gradle/toolchain/GradleDaemonJvmCriteriaDownloadToolchainTest.kt

```diff
@@ -0,0 +1,103 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.idea.gradle.toolchain
+
+import com.intellij.ide.projectWizard.generators.JdkDownloadService
+import com.intellij.openapi.projectRoots.impl.jdkDownloader.JdkDownloadTask
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.ui.TestDialogManager
+import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import com.intellij.testFramework.replaceService
+import com.intellij.util.ui.UIUtil
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.times
+import java.nio.file.Files
+import kotlin.io.path.Path
+
+class GradleDaemonJvmCriteriaDownloadToolchainTest : BasePlatformTestCase() {
+
+    private lateinit var jdkDownloadService: JdkDownloadService
+
+    override fun setUp() {
+        super.setUp()
+        jdkDownloadService = mock(JdkDownloadService::class.java)
+        project.replaceService(JdkDownloadService::class.java, jdkDownloadService, testRootDisposable)
+    }
+
+    fun `test Given unsupported criteria When download matching JDK Then expected exception is thrown`() {
+        lateinit var exceptionMessage: String
+        TestDialogManager.setTestDialog({
+            exceptionMessage = it
+            Messages.OK
+        }, testRootDisposable)
+
+        createDaemonJvmPropertiesFile(0, "unknown vendor")
+        GradleDaemonJvmCriteriaDownloadToolchain.downloadJdkMatchingCriteria(project, project.basePath.orEmpty())
+
+        UIUtil.dispatchAllInvocationEvents()
```

Comment:
### Flaky tests

Could you please explain why you need to dispatch all invocations on the EDT? I don't see this line in other tests in this class. I expect to have it everywhere or nowhere.

Reply:
This is needed because is actually validating an error message displayed to the EDT when trying to download a JDK matching the criteria, for this reason requires to wait for it before to proceed with the assertion. The `UIUtil.dispatchAllInvocationEvents()` is something widely used on the tests when applicable like in this case